From 2deca96cb8836a95095354cc717e1738f10b8ce1 Mon Sep 17 00:00:00 2001
From: Matthew Finkel <sysrqb@torproject.org>
Date: Mon, 1 Feb 2021 21:03:54 +0000
Subject: [PATCH 3/3] Pick up merge conflict resolutions


diff --git a/src/feature/hs/hs_service.c b/src/feature/hs/hs_service.c
index 2f3f45f252..c29f39c6b4 100644
--- a/src/feature/hs/hs_service.c
+++ b/src/feature/hs/hs_service.c
@@ -3192,6 +3192,9 @@ should_service_upload_descriptor(const hs_service_t *service,
   /* Don't upload desc if we don't have a live consensus */
   if (!networkstatus_get_reasonably_live_consensus(now,
                                             usable_consensus_flavor())) {
+    msg = tor_strdup("No reasonably live consensus");
+    log_cant_upload_desc(service, desc, msg,
+                         LOG_DESC_UPLOAD_REASON_NO_LIVE_CONSENSUS);
     goto cannot;
   }
 
diff --git a/src/test/test_hs_service.c b/src/test/test_hs_service.c
index 630cfef1fe..8b94bb6cf1 100644
--- a/src/test/test_hs_service.c
+++ b/src/test/test_hs_service.c
@@ -91,9 +91,10 @@ mock_networkstatus_get_reasonably_live_consensus(time_t now, int flavor)
 }
 
 static networkstatus_t *
-mock_networkstatus_get_live_consensus_null(time_t now)
+mock_networkstatus_get_reasonably_live_consensus_null(time_t now, int flavor)
 {
   (void) now;
+  (void) flavor;
   return NULL;
 }
 
@@ -2554,8 +2555,8 @@ test_cannot_upload_descriptors(void *arg)
   hs_init();
   MOCK(get_or_state,
        get_or_state_replacement);
-  MOCK(networkstatus_get_live_consensus,
-       mock_networkstatus_get_live_consensus);
+  MOCK(networkstatus_get_reasonably_live_consensus,
+       mock_networkstatus_get_reasonably_live_consensus);
 
   dummy_state = or_state_new();
 
@@ -2631,17 +2632,17 @@ test_cannot_upload_descriptors(void *arg)
 
   /* 4. Testing missing live consensus. */
   {
-    MOCK(networkstatus_get_live_consensus,
-         mock_networkstatus_get_live_consensus_null);
+    MOCK(networkstatus_get_reasonably_live_consensus,
+         mock_networkstatus_get_reasonably_live_consensus_null);
     setup_full_capture_of_logs(LOG_INFO);
     run_upload_descriptor_event(now);
     expect_log_msg_containing(
       "Service [scrubbed] can't upload its current descriptor: "
-      "No live consensus");
+      "No reasonably live consensus");
     teardown_capture_of_logs();
     /* Reset. */
-    MOCK(networkstatus_get_live_consensus,
-         mock_networkstatus_get_live_consensus);
+    MOCK(networkstatus_get_reasonably_live_consensus,
+         mock_networkstatus_get_reasonably_live_consensus);
   }
 
   /* 5. Test missing minimum directory information. */
@@ -2680,7 +2681,7 @@ test_cannot_upload_descriptors(void *arg)
  done:
   hs_free_all();
   UNMOCK(count_desc_circuit_established);
-  UNMOCK(networkstatus_get_live_consensus);
+  UNMOCK(networkstatus_get_reasonably_live_consensus);
   UNMOCK(get_or_state);
 }
 
-- 
2.25.1

