diff -ru nss-3.66/nss/build.sh nss-3.66.n/nss/build.sh
--- nss-3.66/nss/build.sh	2021-05-28 09:50:43.000000000 +0000
+++ nss-3.66.n/nss/build.sh	2021-10-15 15:20:52.027557223 +0000
@@ -69,7 +69,7 @@
 ninja_params=()
 
 # Assume that the target architecture is the same as the host by default.
-host_arch=$(python "$cwd/coreconf/detect_host_arch.py")
+host_arch=$(python3 "$cwd/coreconf/detect_host_arch.py")
 target_arch=$host_arch
 
 # Assume that MSVC is wanted if this is running on windows.
