diff -ru nss-3.65/nss/lib/ckfw/builtins/builtins.gyp nss-3.65.n/nss/lib/ckfw/builtins/builtins.gyp
--- nss-3.65/nss/lib/ckfw/builtins/builtins.gyp	2021-09-29 12:48:34.982000000 +0200
+++ nss-3.65.n/nss/lib/ckfw/builtins/builtins.gyp	2021-09-29 12:48:43.152000000 +0200
@@ -30,7 +30,7 @@
         {
           'msvs_cygwin_shell': 0,
           'action': [
-            'python',
+            'python3',
             'certdata.py',
             'certdata.txt',
             '<@(_outputs)',
