diff --git a/components/viaduct/src/backend/ffi.rs b/components/viaduct/src/backend/ffi.rs
index cca6bc68f..5d11eb026 100644
--- a/components/viaduct/src/backend/ffi.rs
+++ b/components/viaduct/src/backend/ffi.rs
@@ -37,6 +37,8 @@ macro_rules! backend_error {
 pub struct FfiBackend;
 impl Backend for FfiBackend {
     fn send(&self, request: crate::Request) -> Result<crate::Response, Error> {
+        return Err(backend_error!("Viaduct is disabled in Tor Browser."));
+
         use ffi_support::IntoFfi;
         use prost::Message;
         super::note_backend("FFI (trusted)");
