diff --git a/libs/build-all.sh b/libs/build-all.sh
index 650c1299..6c4e5404 100755
--- a/libs/build-all.sh
+++ b/libs/build-all.sh
@@ -128,6 +128,15 @@ echo $'\
      fi
 ' | patch "${NSS_SRC_PATH}/nspr/configure"
 
+rm -f python
+ln -s /usr/bin/python3 python
+export PATH=$(pwd):$PATH
+patch_13028=$(realpath bug_13028.patch)
+pushd $NSS_SRC_PATH
+# Apply our proxy bypass defense-in-depth here as well to be on the safe side.
+patch -p2 < $patch_13028
+popd
+
 if [[ "${PLATFORM}" == "ios" ]]
 then
   ./build-all-ios.sh "${SQLCIPHER_SRC_PATH}" "${NSS_SRC_PATH}"
