diff --git a/glean-core/android-native/build.gradle b/glean-core/android-native/build.gradle
index 53716d523..d38af8320 100644
--- a/glean-core/android-native/build.gradle
+++ b/glean-core/android-native/build.gradle
@@ -48,6 +48,8 @@ android {
 }

 cargo {
+    cargoCommand = "${rootDir}/glean-core/android-native/cargo-no-build.sh"
+
     // The directory of the Cargo.toml to build.
     module = '../bundle-android'

diff --git a/glean-core/android/build.gradle b/glean-core/android/build.gradle
index 7748378b4..5af06b1c3 100644
--- a/glean-core/android/build.gradle
+++ b/glean-core/android/build.gradle
@@ -197,7 +197,7 @@ def generateUniffiBindings = tasks.register("generateUniffiBindings") {
     doFirst {
         exec {
             workingDir project.rootDir
-            commandLine 'cargo', 'uniffi-bindgen', 'generate', '--no-format', "${project.projectDir}/${udlFilePath}", '--language', 'kotlin', '--out-dir', UNIFFI_OUT_DIR.get()
+            commandLine '${rootDir}/glean-core/android-native/uniffi-bindgen', 'generate', '--no-format', "${project.projectDir}/${udlFilePath}", '--language', 'kotlin', '--out-dir', UNIFFI_OUT_DIR.get()
         }
     }
     outputs.dir UNIFFI_OUT_DIR
