diff --git a/build.gradle b/build.gradle
index b22a0737..d335aa5b 100644
--- a/build.gradle
+++ b/build.gradle
@@ -39,6 +39,13 @@ buildscript {
     ]
 
     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
         mavenCentral()
         google()
         jcenter()
@@ -88,6 +95,13 @@ apply plugin: 'de.undercouch.download'
 
 allprojects {
     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
         google()
         jcenter()
         maven {
diff --git a/settings.gradle b/settings.gradle
index f652bd02..8c30a368 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,6 +1,19 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pluginManagement {
+    repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
+    }
+}
+
 import org.yaml.snakeyaml.Yaml
 
 includeBuild('tools/nimbus-gradle-plugin') {
@@ -14,6 +27,13 @@ buildscript {
         classpath 'org.yaml:snakeyaml:2.0'
     }
     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
         jcenter()
     }
 }
diff --git a/tools/nimbus-gradle-plugin/settings.gradle b/tools/nimbus-gradle-plugin/settings.gradle
index 7db19be0..1f36b991 100644
--- a/tools/nimbus-gradle-plugin/settings.gradle
+++ b/tools/nimbus-gradle-plugin/settings.gradle
@@ -8,6 +8,13 @@ buildscript {
         classpath 'org.yaml:snakeyaml:2.0'
     }
     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
         jcenter()
     }
 }
