diff --git a/build.gradle b/build.gradle
index bd516e58c..b30d552c9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -14,6 +14,14 @@ buildscript {
     }

     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
+
         mavenCentral()
         google()
         maven {
@@ -59,6 +67,14 @@ plugins {

 allprojects {
     repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
+
         google()
         mavenCentral()
         maven {
diff --git a/settings.gradle b/settings.gradle
index 4117bc0d3..f1b028dad 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -1,6 +1,19 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pluginManagement {
+    repositories {
+        maven {
+            url "file:///var/tmp/dist/gradle-dependencies"
+            metadataSources {
+                gradleMetadata()
+                mavenPom()
+            }
+        }
+    }
+}
+
 import org.yaml.snakeyaml.Yaml

 // We prefer `appServicesRootDir` over `rootDir` to help us on the path to the monorepo.
@@ -23,6 +36,14 @@ buildscript {
     if (!gradle.root.hasProperty("mozconfig")) {
         // in app-services
         repositories {
+            maven {
+                url "file:///var/tmp/dist/gradle-dependencies"
+                metadataSources {
+                    gradleMetadata()
+                    mavenPom()
+                }
+            }
+
             mavenCentral()
         }
     } else {
diff --git a/tools/nimbus-gradle-plugin/settings.gradle b/tools/nimbus-gradle-plugin/settings.gradle
index 7d907f057..c0d213503 100644
--- a/tools/nimbus-gradle-plugin/settings.gradle
+++ b/tools/nimbus-gradle-plugin/settings.gradle
@@ -14,6 +14,14 @@ buildscript {
     if (!gradle.root.hasProperty("mozconfig")) {
         // in app-services
         repositories {
+            maven {
+                url "file:///var/tmp/dist/gradle-dependencies"
+                metadataSources {
+                    gradleMetadata()
+                    mavenPom()
+                }
+            }
+
             mavenCentral()
         }
     } else {
