From 9f848d22882e49d38ff494e684dc54661ec9b714 Mon Sep 17 00:00:00 2001
From: aguestuser <aguestuser@torproject.org>
Date: Mon, 28 Mar 2022 10:43:02 -0400
Subject: [PATCH] Use local maven repository for gradle dependencies

---
 build.gradle    | 2 ++
 settings.gradle | 7 +++++++
 2 files changed, 9 insertions(+)

diff --git a/build.gradle b/build.gradle
index 8e78d9af..635f0740 100644
--- a/build.gradle
+++ b/build.gradle
@@ -38,6 +38,7 @@ buildscript {
     ]
 
     repositories {
+        mavenLocal()
         mavenCentral()
         google()
         jcenter()
@@ -90,6 +91,7 @@ apply plugin: 'de.undercouch.download'
 
 allprojects {
     repositories {
+        mavenLocal()
         google()
         jcenter()
         maven {
diff --git a/settings.gradle b/settings.gradle
index 2349f829..5d9ceb85 100644
--- a/settings.gradle
+++ b/settings.gradle
@@ -2,12 +2,19 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+pluginManagement {
+  repositories {
+    mavenLocal()
+  }
+}
+
 import org.yaml.snakeyaml.Yaml
 buildscript {
     dependencies {
         classpath 'org.yaml:snakeyaml:1.23'
     }
     repositories {
+        mavenLocal()
         jcenter()
     }
 }
-- 
2.32.0

