From 4aa1038fd0d3acc212579fbd94566e062dd187e6 Mon Sep 17 00:00:00 2001
From: Richard Pospesel <richard@torproject.org>
Date: Mon, 22 Apr 2024 17:38:49 +0000
Subject: [PATCH] add WebTunnel Support

--- 
diff --git a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
index 8a439f8..77e6f35 100644
--- a/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
+++ b/universal/src/main/java/com/msopentech/thali/toronionproxy/TorConfigBuilder.java
@@ -512,7 +512,7 @@ public final class TorConfigBuilder {
     }
 
     public TorConfigBuilder transportPlugin(String obfsPath, String snowPath, String conjurePath) {
-        buffer.append("ClientTransportPlugin meek_lite,obfs3,obfs4 exec ").append(obfsPath).append('\n');
+        buffer.append("ClientTransportPlugin meek_lite,obfs3,obfs4,webtunnel exec ").append(obfsPath).append('\n');
         buffer.append("ClientTransportPlugin snowflake exec ").append(snowPath).append('\n');
         buffer.append("ClientTransportPlugin conjure exec ").append(conjurePath).append(" -registerURL https://registration.refraction.network/api\n");
         return this;
