From a39b5c7b089c249f1ba04c297babe22d2bcd75f2 Mon Sep 17 00:00:00 2001
From: Georg Koppen <gk@torproject.org>
Date: Sat, 26 Dec 2020 21:17:59 +0000
Subject: [PATCH] test


diff --git a/android/build.gradle b/android/build.gradle
index a8d9bdc..e0eb541 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -7,6 +7,7 @@ buildscript {
         mavenLocal()
         mavenCentral()
         google()
+        jcenter()
     }
     dependencies {
         classpath "com.android.tools.build:gradle:${androidplugin}"
@@ -14,11 +15,11 @@ buildscript {
 }
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 29
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 28
+        targetSdkVersion 29
         versionCode 1
         versionName "0.0.3"
         consumerProguardFiles 'proguard-rules.pro'
@@ -40,7 +41,9 @@ publishing {
             groupId 'com.msopentech.thali.toronionproxy.android'
             artifactId 'android'
             version '0.0.3'
-            artifact(sourceJar)
+            afterEvaluate {
+              artifact(sourceJar)
+            }
             artifact("$buildDir/outputs/aar/android-release.aar")
             pom.withXml {
                 def dependenciesNode = asNode().appendNode('dependencies')
diff --git a/android_tor_installer/build.gradle b/android_tor_installer/build.gradle
index 554fd49..92e84d4 100644
--- a/android_tor_installer/build.gradle
+++ b/android_tor_installer/build.gradle
@@ -7,6 +7,7 @@ buildscript {
         mavenLocal()
         mavenCentral()
         google()
+        jcenter()
     }
     dependencies {
         classpath "com.android.tools.build:gradle:${androidplugin}"
@@ -14,11 +15,11 @@ buildscript {
 }
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 29
 
     defaultConfig {
         minSdkVersion 16
-        targetSdkVersion 28
+        targetSdkVersion 29
         versionCode 1
         versionName "0.0.3"
 
@@ -48,7 +49,7 @@ repositories {
     mavenLocal()
     mavenCentral()
     google()
-    maven { url "https://repo.spring.io/plugins-release" }
+    jcenter()
     maven { url "https://raw.githubusercontent.com/guardianproject/gpmaven/master" }
 
 }
diff --git a/build.gradle b/build.gradle
index a269024..43b9e13 100644
--- a/build.gradle
+++ b/build.gradle
@@ -8,7 +8,7 @@ subprojects {
   repositories {
     mavenLocal()
     mavenCentral()
-    maven { url "https://repo.spring.io/plugins-release" }
+    jcenter()
   }
 }
 
-- 
2.30.0.rc2

